### PR TITLE
feat(pose_initializer): partial map loading

### PR DESCRIPTION
**Pull request by kminoda**
## Description
Support partial pointcloud map loading for map_height_filter.

Since there is a limit for message data size, current map_height_filter cannot receive pointcloud map larger than ~2GB. In order to address this large size map problem, I propose to incorporate [partial map loading interface](https://github.com/autowarefoundation/autoware.universe/tree/main/map/map_loader#send-partial-pointcloud-map-ros-2-service). 

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by isamu-takagi** [comment]
@kminoda Could not initialize the pose if partial_map_load_enabled was true. Are there any other settings required?

**Comment by kminoda** [comment]
@isamu-takagi Thank you for testing.
Yes, you also have to change the setting in pointcloud_map_loader. I added some description to clarify how to use the interface.

File: localization/pose_initializer/config/pose_initializer.param.yaml [hunk 1]
```diff
@@ -22,3 +22,7 @@
         0.0, 0.0, 0.0,  0.0,  0.01, 0.0,
         0.0, 0.0, 0.0,  0.0,  0.0,  0.2,
       ]
+
+    # use partial map loading for large maps
+    # Note: You also need to enable partial_map_loading interface in pointcloud_map_loader to use this functionality
+    enable_partial_map_load: true
```
Comment (isamu-takagi) [inline localization/pose_initializer/config/pose_initializer.param.yaml:28]: Maybe it was overwritten when adding a comment.
```suggestion
    enable_partial_map_load: false
```

Comment (kminoda) [reply localization/pose_initializer/config/pose_initializer.param.yaml]: Thank you! I resolved here: https://github.com/autowarefoundation/autoware.universe/pull/2500/commits/5dbde23be17426ba393dfe3ba0f8c9e2cbedb2e9

**Review by isamu-takagi — APPROVED** [review]
LGTM

**Pull request merged**